spec/java/util/HashSet.main.lsl

```diff
@@ -127,10 +128,7 @@ automaton HashSetAutomaton
 
     proc _generateKey (visitedKeys: map<Object, Object>): Object
     {
-        result = action SYMBOLIC("java.lang.Object");
-        action ASSUME(result != null);
-        val isKeyExist: boolean = action MAP_HAS_KEY(this.storage, result);
-        action ASSUME(isKeyExist);
+        result = action MAP_GET_ANY_KEY(this.storage);
 
         val isKeyWasVisited: boolean = action MAP_HAS_KEY(visitedKeys, result);
         action ASSUME(!isKeyWasVisited);
```

Comment:
Consider using supplementary `unseen: map<Object, Object>` to store items for later retrieval as suggested by Valentin.

Reply:
Fixed